File: state/Desafio/Camila Aparecida_DesafioState/DesafioState/src/desafiostate/Eduardo Junior - DesafioState/src/MarioGrande.java
public class MarioGrande implements MarioState {

    @Override
    public MarioState pegarCogumelo() {
        System.out.println("Ganha 1000 pontos");
        return this;
    }

    @Override
    public MarioState pegarFlor() {
        System.out.println("Mario Fogo");
        return new MarioFogo();
    }

    @Override
    public MarioState pegarPena() {
        System.out.println("Mario Capa");
        return new MarioCapa();
    }

    @Override
    public MarioState levarDano() {
        System.out.println("Mario Pequeno");
        return new MarioPequeno();
    }  
}

File: state/Desafio/Renny/PadraoState/Entitades/MarioGrande.cs
using PadraoState.Interfaces;
using System;

namespace PadraoState.Entitades
{
    class MarioGrande : MarioState
    {
        public MarioGrande()
        {
            Console.WriteLine("Mario Grande");
        }

        public MarioState LevarDano()
        {
            return new MarioPequeno();
        }

        public MarioState PegarCogumelo()
        {
            Console.Write("Pontos +1000 - ");
            return new MarioGrande();
        }

        public MarioState PegarFlor()
        {
            return new MarioFogo();
        }

        public MarioState PegarPena()
        {
            return new MarioPena();
        }
    }
}
